cli/azd/pkg/project/service_manager.go

```diff
@@ -658,6 +684,59 @@ func runCommand[T any](
 	return result, err
 }
 
+// getTargetResourceForService resolves the target resource for a service configuration.
+// For DotNetContainerAppTarget, it handles container app environment resolution.
+// For other service types, it delegates to the resource manager.
+func (sm *serviceManager) getTargetResourceForService(
+	ctx context.Context,
+	serviceConfig *ServiceConfig,
+) (*environment.TargetResource, error) {
+	if serviceConfig.Host == DotNetContainerAppTarget {
```

Comment:
Would love to move aspire to its own extension (someday) with its own host/service target so it can just do things it cares about instead of all the special handling throughout `azd`.